app/main.py

```diff
@@ -1,3 +1,68 @@
 class Distance:
     # Write your code here
-    pass
+    def __init__(self, km: (int, float)):
+        self.km = km
+
+    def __add__(self, other):
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __mul__(self, other):
+        if isinstance(other, Distance):
+            return Distance(self.km * other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km * other)
+
+    def __iadd__(self, other):
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
```

Comment:
it would be better to use only `else` here

Reply:
Done